lib/phoenix_live_view/html_engine.ex

```diff
@@ -762,4 +765,50 @@ defmodule Phoenix.LiveView.HTMLEngine do
   defp void?(_), do: false
 
   defp to_location(%{line: line, column: column}), do: [line: line, column: column]
+
+  # Check if `phx-update` or `phx-hook` is present in attrs and raises in case
+  # there is no ID attribute set.
+  defp validate_phx_attrs!(attrs, meta, state),
+    do: validate_phx_attrs!(attrs, meta, state, nil, false)
+
+  defp validate_phx_attrs!([], meta, state, attr, false)
+       when attr in ["phx-update", "phx-hook"] do
+    message = "attribute \"#{attr}\" requires the \"id\" attribute to be set"
+
+    raise ParseError,
+      line: meta.line,
+      column: meta.column,
+      file: state.file,
+      description: message
+  end
+
+  defp validate_phx_attrs!([], _meta, _state, _attr, _id?), do: :ok
+
+  # Handle <div phx-update="ignore" {@some_var}>Content</div> since here the ID
+  # might be inserted dinamically so we can't raise at compile time.
+  defp validate_phx_attrs!([{:root, _} | _t], _meta, _state, _attr, _id?),
```

Comment:
@josevalim I guess that's enough to check for dynamic vars? Let me know otherwise 🙌🏻